libbeat/tests/integration/http_test.go

```diff
@@ -57,12 +57,14 @@ output.console:
 	mockbeat.WriteConfigFile(cfg)
 	mockbeat.Start()
 	mockbeat.WaitForLogs("Starting stats endpoint", 60*time.Second)
+	time.Sleep(time.Second)
 
-	r, err := http.Get("http://localhost:5066")
+	r, err := http.Get("http://localhost:5066") //nolint:noctx // fine for tests
 	require.NoError(t, err)
 	require.Equal(t, http.StatusOK, r.StatusCode, "incorrect status code")
```

Comment:
[Blocker] cont.

you could use it instead

```suggestion
buff := &bytes.Buffer{}
	require.Eventually(t, func() bool {
		buff.Reset()
		r, err := http.Get("http://localhost:5066")
		if err != nil {
			_, _ = fmt.Fprintf(buff, "stats endpoint not available: %v", err)
			return false
		}
		if r.StatusCode != http.StatusOK {
			_, _ = fmt.Fprintf(buff, "stats endpoint: bad HTTPnstatus: %s",
				r.Status)
			return false
		}
		return true
	}, time.Second, 100*time.Millisecond,
		"stats endpoint never become available: %s", buff)
```

if you want, you could even remove the `WaitForLogs`. Just don't mix `WaitForLogs` and `Eventually` because `WaitForLogs` uses eventually underneath.